app/src/renderer/js/export.js

```diff
@@ -0,0 +1,52 @@
+import {ipcRenderer, remote, dialog} from 'electron';
+
+import {handleTrafficLightsClicks} from '../js/utils';
+
+let isExportInProgress = false;
+
+window.onbeforeunload = e => {
+  if (isExportInProgress) {
+    const buttonIndex = dialog.showMessageBox(remote.getCurrentWindow(), {
+      type: 'question',
+      buttons: ['Cancel Export', 'Continue'],
+      defaultId: 1,
+      message: 'Are you sure you want to cancel exporting?',
+      detail: 'It will not be saved'
+    });
+    if (buttonIndex === 0) {
+      ipcRenderer.send('cancel-export');
+    } else {
+      e.returnValue = true; // Prevents closing
```

Comment:
@sindresorhus HELP! This doesn't work. If this `e.returnValue = true` is outside of this if-else statement, it will prevent closing. But now, the window closes in both cases... do you have any idea whats happening here?

Reply:
never mind. fixed.